app/application.js

```diff
@@ -1,18 +1,53 @@
 import Mn from 'backbone.marionette';
 import Backbone from 'backbone';
-import tracksView from 'views/tracks';
+import Tracks from './collections/tracks';
+import Playlists from './collections/playlists';
+import AppLayout from './views/app_layout';
 
+require('./styles/app.styl');
 
-class Application extends Mn.Application {
+const Application = Mn.Application.extend({
 
-    onStart () {
-        console.log('start');
+    initialize: function () {
+        this.allTracks = new Tracks([], { type: 'all' });
+        this.upNext = new Tracks();
+        this.upNext.on('change:currentTrack', function() {
+            const upNext = app.upNext;
+            const currentTrack = upNext.getAttr('currentTrack');
+            if (!upNext.contains(currentTrack)) {
+                upNext.push(currentTrack);
+            }
+        });
+        this.allTracks.fetch({
+            success: function() {
+                app.allTracks.each(function(track) {
+                    app.upNext.add(track);
+                });
+            }
+        });
+
+        this.allPlaylists = new Playlists();
+        this.allPlaylists.fetch();
+
+        this.headerInfos = new Backbone.Model({
```

Comment:
I can see what you want top do with it, but I propose you use a top-level model called `appState` which's here to handle the state of the app, like the title of current app, the number of tracks in current playlist, etc. It'll be more flexible that an more opinionated model dedicated to a component (like `headerInfos`).